Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/Visuals/TranslationHandles.cs

```diff
@@ -0,0 +1,338 @@
+            config.colliderTypeChanged.RemoveListener(UpdateColliderType);
+        }
+
+        protected void UpdateColliderType()
+        {
+            foreach (var handle in handles)
+            {
+                // remove old colliders
+                bool shouldCreateNewCollider = false;
+                var oldBoxCollider = handle.GetComponent<BoxCollider>();
+                if (oldBoxCollider != null && config.TranslationHandlePrefabColliderType == HandlePrefabCollider.Sphere)
+                {
+                    shouldCreateNewCollider = true;
+                    Object.Destroy(oldBoxCollider);
+                }
+
+                var oldSphereCollider = handle.GetComponent<SphereCollider>();
+                if (oldSphereCollider != null && config.TranslationHandlePrefabColliderType == HandlePrefabCollider.Box)
+                {
+                    shouldCreateNewCollider = true;
+                    Object.Destroy(oldSphereCollider);
+                }
+
+                if (shouldCreateNewCollider)
+                {
+                    // attach new collider
+                    var handleBounds = VisualUtils.GetMaxBounds(GetVisual(handle).gameObject);
+                    var invScale = handleBounds.size.x == 0.0f ? 0.0f : config.HandleSize / handleBounds.size.x;
+                    Vector3 colliderSizeScaled = handleBounds.size * invScale;
+                    Vector3 colliderCenterScaled = handleBounds.center * invScale;
+                    if (config.TranslationHandlePrefabColliderType == HandlePrefabCollider.Box)
+                    {
+                        BoxCollider collider = handle.gameObject.AddComponent<BoxCollider>();
+                        collider.size = colliderSizeScaled;
+                        collider.center = colliderCenterScaled;
+                        collider.size += config.ColliderPadding;
+                    }
+                    else
+                    {
+                        SphereCollider sphere = handle.gameObject.AddComponent<SphereCollider>();
+                        sphere.center = colliderCenterScaled;
+                        sphere.radius = colliderSizeScaled.x * 0.5f;
+                        sphere.radius += VisualUtils.GetMaxComponent(config.ColliderPadding);
+                    }
+                }
+
+            }
+        }
+
+        internal const int NumFaces = 6;
+
+        private Vector3[] faceCenters = new Vector3[NumFaces];
+        private CardinalAxisType[] faceAxes;
+
+        internal int GetTranslationHandleIdx(Transform handle)
+        {
+            for (int i = 0; i < handles.Count; ++i)
+            {
+                if (handle == handles[i])
+                {
+                    return i;
+                }
+            }
+
+            return handles.Count;
+        }
+
+        internal Vector3 GetFaceCenter(int index)
+        {
+            Debug.Assert(index >= 0 && index <= NumFaces, "Face center index out of bounds");
```

Comment:
```suggestion
            Debug.Assert(index >= 0 && index < NumFaces, "Face center index out of bounds");
```

Reply:
Classic off by one

Reply:
Fixed in the other handles, too. Thanks for the catch.